src/utils.ts

```diff
@@ -222,3 +222,16 @@ export function getGlobalCSSVariable(name: string) {
     `--${name}`,
   );
 }
+
+export const getPixelRatio = (): number =>
+  typeof window !== "undefined" && window.devicePixelRatio
+    ? window.devicePixelRatio
+    : 1;
+
+export function createCanvasObject() {
+  if (typeof document !== "undefined" && document.createElement) {
+    return document.createElement("canvas");
+  }
+
+  return require("canvas").createCanvas();
```

Comment:
Can you check what happens in the generated bundle for the browser. Does it include the whole canvas library? How big is it?

Reply:
Instead of CJS, let's use dynamic `import()`, which will definitely code-split it into separate bundle.

Reply:
Does dynamic import work on nodejs?

Reply:
Well, on node, even non-dynamic import won't work on pre-14 (13 too?) versions, so we'll need transpilation either way.

Reply:
What do you mean by "non-dynamic import"? Putting require inside a function just works fine. require is just a regular synchronous function in node. In the browser we need to know statically so we can extract all the dependencies at bundle time.

Reply:
By non-dynamic import, I mean `import * from ...`. It's still ESM and Node can't parse that until v14, and even there it only works with ESM modules.

As it stands, we gain no benefit mixing CJS and ESM.